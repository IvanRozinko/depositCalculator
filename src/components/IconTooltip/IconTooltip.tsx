import React, { useState } from 'react'
import { Tooltip, makeStyles } from '@material-ui/core';
import Help from '@material-ui/icons/HelpOutline';

const useStyle = makeStyles({
  ml: {
    marginLeft: 8,
    fontSize: 20,
  }
})
function IconTooltip() {
  const [ open, setOpen ] = useState(false);
  const styles = useStyle();

  const handleClick = () => {
    if (!open) {
      setOpen(true);
      setTimeout( () => setOpen(false), 1000);
    }
  }
  return (
    <Tooltip
      open={open}
      title='Загальна ставка оподаткування 19,5 % (18% податок + 1,5 % військовий збір)'
    >
      <Help 
        className={styles.ml}
        color='primary'
        onClick={handleClick}
      />
    </Tooltip>
  )
}

export default IconTooltip
