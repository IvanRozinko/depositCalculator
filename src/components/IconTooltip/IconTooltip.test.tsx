import IconTooltip from "./IconTooltip";
import { shallow } from 'enzyme';
import { Tooltip } from '@material-ui/core';
import Help from '@material-ui/icons/HelpOutline';

describe('<IconTooltip /> :', () => {
  const wrapper = shallow(
    <IconTooltip />
  )
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should shows up on icon click', () => {
    wrapper.find(Help).simulate('click');
    expect(wrapper.find(Tooltip).props().open).toBe(true)
  })
})
