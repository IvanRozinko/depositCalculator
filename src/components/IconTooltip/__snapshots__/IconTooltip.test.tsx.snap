// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconTooltip /> : should render correctly 1`] = `
<WithStyles(ForwardRef(Tooltip))
  open={false}
  title="Загальна ставка оподаткування 19,5 % (18% податок + 1,5 % військовий збір)"
>
  <Memo(ForwardRef(HelpOutlineIcon))
    className="makeStyles-ml-1"
    color="primary"
    onClick={[Function]}
  />
</WithStyles(ForwardRef(Tooltip))>
`;
